src/flag_gems/ops/repeat_interleave.py

```diff
@@ -0,0 +1,64 @@
+import torch
+import triton
+
+from ..utils.pointwise_dynamic import pointwise_dynamic
+from ..utils.shape_utils import c_contiguous_stride, volume
+from ..utils.tensor_wrapper import StridedBuffer
+
+
+@pointwise_dynamic(num_inputs=1, promotion_methods=[(0, "DEFAULT")])
+@triton.jit
+def copy_func(x):
+    return x
+
+
+def repeat_interleave_self_int(inp, repeats, dim=None, *, output_size=None):
+    if dim is None:
+        nelems = volume(inp.shape)
+        inp_shape = [
+            nelems,
+        ]
+        inp_stride = [
+            1,
+        ]
+        output_shape = [
+            nelems,
+        ]
+        dim = 0
```

Comment:
When dim is None, the input is treated like a 1d vector. But it does not mean that it **is** a tensor with stride 1.

The input tensor may be possible to be viewed as a 1d tensor, but not with stride 1. The input tensor may not be possible to be viewed as a 1d tensor, in which cases it has to be flattened beforehand.

Note that `torch.view` never copies data, while `flatten, reshape` may copy data when it it impossible to reinterpret the tensor with another shape without copying data.

Test with the following cases:

```python
x = torch.randn(4, 6, device="cuda")[::2, ::2]
y1 = torch.repeat_interleave(x, 2)
y2 = flag_gems.repeat_interleave_self_int(x, 2)
```

or 

```python
x = torch.randn(10, device="cuda")[::2]
y1 = torch.repeat_interleave(x, 2)
y2 = flag_gems.repeat_interleave_self_int(x, 2)
```

And we can see that this implementation cannot handle non-contiguous input.

Also, add soeme tests to ensure that in these cases, the results are correct.

Reply:
According to the description of torch API reference: dim is "The dimension along which to repeat values. By default, use the flattened input array, and return a flat output array."

So, used `inp.flatten()` to resolve this test case.